recipes/libcurl/all/conanfile.py

```diff
@@ -207,13 +205,24 @@ def build_requirements(self):
         else:
             self.build_requires("libtool/2.4.6")
             self.build_requires("pkgconf/1.7.4")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-                self.build_requires("msys2/cci.latest")
+            if self._settings_build.os == "Windows" and not get_env("CONAN_BASH_PATH") and self.conf.get("tools.microsoft.bash:path"):
```

Comment:
```suggestion
            if self._settings_build.os == "Windows" and not get_env("CONAN_BASH_PATH") and self.conf.get("tools.microsoft.bash:path"):
```

Not sure to understand the condition `and self.conf.get("tools.microsoft.bash:path")`. If not defined, we don't add msys2 to tool_requires? Seems broken logic.

Reply:
Shall I remove it ?

Reply:
I mean that it should be `not self.conf.get("tools.microsoft.bash:path")`, don't you think?

Reply:
I update it accordingly. It makes senses indeed.